app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -0,0 +1,362 @@
+        mapboxNavigation.registerRouteProgressObserver(replayProgressObserver)
+        mapboxNavigation.registerRoutesObserver(routesObserver)
+        mapboxNavigation.registerRouteAlternativesObserver(alternativesObserver)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapboxNavigation.unregisterRouteProgressObserver(replayProgressObserver)
+        mapboxNavigation.unregisterLocationObserver(locationObserver)
+        mapboxNavigation.unregisterRoutesObserver(routesObserver)
+        mapboxNavigation.unregisterRouteAlternativesObserver(alternativesObserver)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        routeLineApi.cancel()
+        routeLineView.cancel()
+        mapboxReplayer.finish()
+        mapboxNavigation.onDestroy()
+    }
+
+    /**
+     * Request routes between the two points.
+     */
+    private fun findRoute(origin: Point?, destination: Point?) {
+        val routeOptions = RouteOptions.builder()
+            .applyDefaultNavigationOptions()
+            .applyLanguageAndVoiceUnitOptions(this)
+            .coordinatesList(listOf(origin, destination))
+            .layersList(listOf(mapboxNavigation.getZLevel(), null))
+            .alternatives(true) // make sure you set the `alternatives` flag to true in route options
+            .build()
+        mapboxNavigation.requestRoutes(
+            routeOptions,
+            object : NavigationRouterCallback {
+                override fun onRoutesReady(
+                    routes: List<NavigationRoute>,
+                    routerOrigin: RouterOrigin
+                ) {
+                    if (routes.isNotEmpty()) {
+                        mapboxNavigation.setNavigationRoutes(routes)
+                    }
+                }
+
+                override fun onFailure(
+                    reasons: List<RouterFailure>,
+                    routeOptions: RouteOptions
+                ) {
+                    // no impl
+                }
+
+                override fun onCanceled(routeOptions: RouteOptions, routerOrigin: RouterOrigin) {
+                    // no impl
+                }
+            }
+        )
+    }
+
+    private fun updateCamera(point: Point, bearing: Double? = null) {
+        val mapAnimationOptions = MapAnimationOptions.Builder().duration(1500L).build()
+        binding.mapView.camera.easeTo(
+            CameraOptions.Builder()
+                .center(point)
+                .bearing(bearing)
+                .zoom(15.0)
+                .pitch(45.0)
+                .padding(EdgeInsets(1000.0, 0.0, 0.0, 0.0))
+                .build(),
+            mapAnimationOptions
+        )
+    }
+
+    private fun replayOriginLocation() {
+        mapboxReplayer.pushEvents(
+            listOf(
+                ReplayRouteMapper.mapToUpdateLocation(Date().time.toDouble(), originPoint)
+            )
+        )
+        mapboxReplayer.playFirstLocation()
+        mapboxReplayer.playbackSpeed(3.0)
+    }
```

Comment:
Instead of using the `ReplayLocationEngine` passed through the `NavigationOptions`, there is new approach. Use the `MapboxNavigation.startReplayTripSession` instead of `startTripSession`.

> do you know a better way to replay original location until ReplayProgressObserver start simulation on the first route progress update.

@VysotskiVadim This was considered and we could still add it https://github.com/mapbox/mapbox-navigation-android/issues/3185. The only reason we don't do it, is because a single location will typically identify stale location issues. So if you want to add a location repeater, we should be able to toggle it on and off.